tests/linearizability/model.go

```diff
@@ -71,39 +102,79 @@ var etcdModel = porcupine.Model{
 		return ok, string(data)
 	},
 	DescribeOperation: func(in, out interface{}) string {
-		request := in.(EtcdRequest)
-		response := out.(EtcdResponse)
-		switch request.Op {
-		case Get:
-			if response.Err != nil {
-				return fmt.Sprintf("get(%q) -> %q", request.Key, response.Err)
-			} else {
-				return fmt.Sprintf("get(%q) -> %q, rev: %d", request.Key, response.GetData, response.Revision)
-			}
-		case Put:
-			if response.Err != nil {
-				return fmt.Sprintf("put(%q, %q) -> %s", request.Key, request.PutData, response.Err)
-			} else {
-				return fmt.Sprintf("put(%q, %q) -> ok, rev: %d", request.Key, request.PutData, response.Revision)
-			}
-		case Delete:
-			if response.Err != nil {
-				return fmt.Sprintf("delete(%q) -> %s", request.Key, response.Err)
-			} else {
-				return fmt.Sprintf("delete(%q) -> ok, rev: %d deleted:%d", request.Key, response.Revision, response.Deleted)
-			}
-		case Txn:
-			if response.Err != nil {
-				return fmt.Sprintf("txn(if(value(%q)=%q).then(put(%q, %q)) -> %s", request.Key, request.TxnExpectData, request.Key, request.TxnNewData, response.Err)
-			} else {
-				return fmt.Sprintf("txn(if(value(%q)=%q).then(put(%q, %q)) -> %v, rev: %d", request.Key, request.TxnExpectData, request.Key, request.TxnNewData, response.TxnSucceeded, response.Revision)
-			}
-		default:
-			return "<invalid>"
-		}
+		return describeEtcdRequestResponse(in.(EtcdRequest), out.(EtcdResponse))
 	},
 }
 
+func describeEtcdRequestResponse(request EtcdRequest, response EtcdResponse) string {
+	prefix := describeEtcdOperations(request.Ops)
+	if len(request.Conds) != 0 {
+		prefix = fmt.Sprintf("if(%s).then(%s)", describeEtcdConditions(request.Conds), prefix)
+	}
+
+	return fmt.Sprintf("%s -> %s", prefix, describeEtcdResponse(request.Ops, response))
+}
+
+func describeEtcdConditions(conds []EtcdCondition) string {
+	opsDescription := make([]string, len(conds))
+	for i := range conds {
+		opsDescription[i] = fmt.Sprintf("%s==%q", conds[i].Key, conds[i].ExpectedValue)
+	}
+	return strings.Join(opsDescription, ", ")
```

Comment:
If you go for golang/c++ like syntax, maybe && to express logical end is more natural

Reply:
Done